lib/volk_rank_archs.c

```diff
@@ -34,13 +34,16 @@ int volk_get_index(const char* impl_names[], // list of implementations by name
 {
     unsigned int i;
     for (i = 0; i < n_impls; i++) {
-        if (!strncmp(impl_names[i], impl_name, 20)) {
+        if (!strncmp(impl_names[i], impl_name, 42)) {
```

Comment:
since `strncmp` returns `int` of "less than, equal to, or greater than zero" can we go with `strncmp () == 0` here? Compilers will know how to optimize the code here, and this avoids any possible issues with binary not (`!`) and negative integers. Yes this code should work fine, but it's just as easy to compare with 0.

Reply:
I really hope it is `!= 0`. Fixed it.